catroid/src/main/java/org/catrobat/catroid/stage/StageListener.java

```diff
@@ -488,31 +458,31 @@ public void render() {
 			}
 			scene.firstStart = false;
 		}
-		if (!paused) {
-			float deltaTime = Gdx.graphics.getDeltaTime();
-
-			float optimizedDeltaTime = deltaTime / deltaActionTimeDivisor;
-			long timeBeforeActionsUpdate = SystemClock.uptimeMillis();
-			while (deltaTime > 0f) {
-				physicsWorld.step(optimizedDeltaTime);
-				stage.act(optimizedDeltaTime);
-				deltaTime -= optimizedDeltaTime;
+		synchronized (this) {
```

Comment:
From my understanding, concurrency problems with the PenActor occur because `PenActor.dispose` is called in `StageListener.finish` (which isn't called from the OpenGL thread).
Dispose the `PenActor` in the `StageListener.dispose` method and you don't need to synchronize the `PenActor`.